Bridges/HttpKernel.php

```diff
@@ -238,14 +239,33 @@ protected function mapResponse(HttpResponse $reactResponse, SymfonyResponse $syR
             $headers['Set-Cookie'] = $cookies;
         }
 
-        $reactResponse->writeHead($syResponse->getStatusCode(), $headers);
+        if ($syResponse instanceof SymfonyStreamedResponse) {
+            $reactResponse->writeHead($syResponse->getStatusCode(), $headers);
+
+            // asynchronously get content
+            ob_start(function($buffer) use ($reactResponse) {
+                $reactResponse->write($buffer);
+                return '';
+            }, 4096);
+
+            $syResponse->sendContent();
 
-        $stdOut = '';
-        while ($buffer = @ob_get_clean()) {
-            $stdOut .= $buffer;
+            // flush remaining content
+            @ob_end_flush();
+            $reactResponse->end();
         }
+        else {
+            ob_start();
+            $content = $syResponse->getContent();
+            @ob_end_flush();
 
-        $reactResponse->end($stdOut . $content);
+            if (!isset($headers['Content-Length'])) {
+                $headers['Content-Length'] = strlen($content);
```

Comment:
What if we have utf8 content? Regarding http spec, content-length is not mandatory and the client reads until connection drops. I'd go the safe path and don't set content-length for now. Did you experience issues with not setting it?